Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs

```diff
@@ -124,22 +124,34 @@ public SandboxedProcessMacKext(SandboxedProcessInfo info, bool ignoreReportedAcc
                     BoundedCapacity = DataflowBlockOptions.Unbounded,
                     MaxDegreeOfParallelism = 1, // Must be one, otherwise SandboxedPipExecutor will fail asserting valid reports
                 });
+
+            // install a 'ProcessStarted' handler that informs the kext of the newly started process
+            ProcessStarted += () => OnProcessStartedAsync().GetAwaiter().GetResult();
         }
 
         /// <inheritdoc />
-        public override void Start() => StartAsync().GetAwaiter().GetResult();
-
-        private async Task StartAsync()
+        protected override System.Diagnostics.Process CreateProcess()
         {
-            base.CreateAndSetUpProcess();
+            var process = base.CreateProcess();
 
-            Process.StartInfo.FileName = "/bin/sh";
-            Process.StartInfo.Arguments = string.Empty;
-            Process.StartInfo.RedirectStandardInput = true;
-            Process.Start();

```

Comment:
The main problem was that this class used to start the process (call `Process.Start()`) while the base class used `m_processExecutor` for other things (like wait for the process to exit, compute process start/exit time, etc.).  Now all that process lifetime management goes through the `AsyncProcessExecutor`